src/OpenSSL/SSL.py

```diff
@@ -701,6 +711,35 @@ def set_default_verify_paths(self):
         """
         set_result = _lib.SSL_CTX_set_default_verify_paths(self._context)
         _openssl_assert(set_result == 1)
+        num = self._check_num_store_objects()
```

Comment:
The env var behavior is still present and functional because we call `SSL_CTX_set_default_verify_paths` before we try the fallback, but we need to fix the lazy loading issue.

Do you know of a way to force the X509_STORE to load the certs? That seems like an easier path (if possible) than trying to replicate the logic.